fiat-roles/src/main/groovy/com/netflix/spinnaker/fiat/permissions/RedisPermissionsRepository.java

```
@@ -16,7 +16,228 @@
+
+  @Override
+  public PermissionsRepository put(@NonNull UserPermission permission) {
+    try (Jedis jedis = jedisSource.getJedis()) {
+      for (Resource r : Resource.values()) {
+        Map<String, String> resourceValues = new HashMap<>();
+
+        Consumer<Named> putInValuesMap = namedResource -> {
+          try {
+            resourceValues.put(namedResource.getName(),
+                               objectMapper.writeValueAsString(namedResource));
+          } catch (JsonProcessingException jpe) {
+            log.error("Serialization exception writing " + permission.getId() + " entry.", jpe);
+          }
+        };
+
+        switch (r) {
+          case ACCOUNT:
+            permission.getAccounts().stream().forEach(putInValuesMap);
+            break;
+          case APPLICATION:
+            permission.getApplications().stream().forEach(putInValuesMap);
+            break;
+        }
+
+        if (!resourceValues.isEmpty()) {
+          jedis.hmset(userKey(permission.getId(), r), resourceValues);
+          jedis.sadd(allUsersKey(), permission.getId());
+        }
+      }
+    } catch (Exception e) {
+      log.error("Storage exception writing " + permission.getId() + " entry.", e);
+    }
+    return this;
+  }
+
+  @Override
+  public UserPermission get(@NonNull String id) {
+    UserPermission userPermission = new UserPermission().setId(id);
+    try (Jedis jedis = jedisSource.getJedis()) {
+      for (Resource r : Resource.values()) {
+        Map<String, String> resourceMap = jedis.hgetAll(userKey(id, r));
+
+        switch (r) {
+          case ACCOUNT:
+            userPermission.getAccounts().addAll(extractAccounts(resourceMap));
+            break;
+          case APPLICATION:
+            userPermission.getApplications().addAll(extractApplications(resourceMap));
+            break;
+        }
+      }
+    } catch (Exception e) {
+      log.error("Storage exception reading " + id + " entry.", e);
```

should this throw something on read error rather than returning null? if it should return null should it return `Optional<UserPermission>` instead?